pep-XXXX.rst

```diff
@@ -0,0 +1,258 @@
+versions of certain core libraries like ``glibc``.
+
+
+External Shared Libraries
+-------------------------
+
+Most desktop and server linux distributions come with a system package manager
+(examples include ``APT`` on Debian-based systems, ``yum`` on
+``RPM``-based systems, and ``pacman`` on Arch linux) that manages, among other
+responsibilities, the installation of shared libraries installed to system
+directories such as ``/usr/lib``. Most non-trivial Python extensions will depend
+on one or more of these shared libraries, and thus function properly only on
+systems where the user has the proper libraries (and the proper
+versions thereof), either installed using their package manager, or installed
+manually by setting certain environment variables such as ``LD_LIBRARY_PATH``
+to notify the runtime linker of the location of the depended-upon shared
+libraries.
+
+
+Versioning of Core Shared Libraries
+-----------------------------------
+
+Even if author or maintainers of a Python extension module with to use no
+external shared libraries, the modules will generally have a dynamic runtime
+dependency on the GNU C library, ``glibc``. While it is possible, statically
+linking ``glibc`` is usually a bad idea because of bloat, and because certain
+important C functions like ``dlopen()`` cannot be called from code that
+statically links ``glibc``. A runtime shared library dependency on a
+system-provided ``glibc`` is unavoidable in practice.
+
+The maintainers of the GNU C library follow a strict symbol versioning scheme
+for backward compatibility. This ensures that binaries compiled against an older
+version of ``glibc`` can run on systems that have a newer ``glibc``. The
+opposite is generally not true -- binaries compiled on newer Linux
+distributions tend to rely upon versioned functions in glibc that are not
+available on older systems.
+
+This generally prevents built distributions compiled on the latest Linux
+distributions from being portable.
+
+
+The ``manylinux_1`` policy
+==========================
+
+For these reasons, to achieve broad portability, Python wheels
+
+ * should depend only on an extremely limited set of external shared
+   libraries; and
+ * should depend only on ``old`` symbol versions in those external shared
+   libraries.
+
+The ``manylinux_1`` policy thus encompasses a standard for what the
+permitted external shared libraries a wheel may depend on, and the maximum
+depended-upon symbol versions therein.
+
+The permitted external shared libraries are: ::
+
+  "libpanelw.so.5", "libncursesw.so.5", "libgcc_s.so.1", "libstdc++.so.6",
+  "libm.so.6", "libdl.so.2", "librt.so.1", "libcrypt.so.1", "libc.so.6",
+  "libnsl.so.1", "libutil.so.1", "libpthread.so.0", "libX11.so.6",
+  "libXext.so.6", "libXrender.so.1", "libICE.so.6", "libSM.so.6",
+  "libGL.so.1", "libgobject-2.0.so.0", "libgthread-2.0.so.0",
+  "libglib-2.0.so.0", "libgmodule-2.0.so.0", and "libgio-2.0.so.0".
```

Comment:
1) Can we format these as big vertical lists like
```
  libpanelw.so.5
  libncursesw.so.5
  ...
```
etc. without the quotes and commas? It's way more friendly for cut-and-pasting that way, and we're not going to run out of vertical space :-)

2) Do we really want to include `libgmodule-2.0.so.0` and `libgio-2.0.so.0`? The way they get thrown in at the end is triggering the same instinct that makes me twitch when I see "one little added feature" thrown into a final release after being left out of the betas, because *surely* it won't cause any problems :-). The fact that these seem to be almost or totally unused by Anaconda/Canopy suggests that no-one will miss them if we leave them out. Also we don't actually know whether they'll work (they'll probably be present, but for all we know libgio has broken ABI recently, we just don't have data). And also it's a lot easier rhetorically if we can say "everything in this list is used by both Anaconda and Enthought", versus "everything in this list is used by both Anaconda and Enthought [except ...]"?